cybershell/memory.py

```
@@ -0,0 +1,74 @@
+        dot = sum(x*y for x,y in zip(a,b))
+        na = (sum(x*x for x in a) ** 0.5) or 1.0
+        nb = (sum(x*x for x in b) ** 0.5) or 1.0
+        return dot/(na*nb)
+
+class NeuralMemory:
+    def __init__(self):
+        self.items: List[MemoryItem] = []
+        self.vec = SimpleVectorizer()
+        self._matrix: List[List[float]] = []
+
+    def add(self, item: MemoryItem):
```

Ruff (0.12.2)

Missing return type annotation for public function `add`

Add return type annotation: `None`

(ANN201)